torch/onnx/_internal/exporter.py

```diff
@@ -915,10 +915,17 @@ def export(self) -> ExportOutput:
                 op_level_debug=self.options.op_level_debug,
             )
 
+            # NOTE: Filter out the initializers with fake tensors, otherwise,
+            # the ONNX exporter will fail: RuntimeError: basic_string::_M_construct null not valid
+            initializers_with_real_tensors: Dict[str, torch.Tensor] = {}
+            for initializer_name, initializer in onnxscript_graph.initializers.items():
+                if not isinstance(initializer, torch._subclasses.FakeTensor):
+                    initializers_with_real_tensors[initializer_name] = initializer
+            onnxscript_graph.initializers = initializers_with_real_tensors
```

Comment:
I would just call add for each initializer

Reply:
They are already in it. We just need another iteration to filter out the ones with fake tensor before sending them to `to_model_proto`.

Reply:
Currently, they are all filtered out by `include_initializers`, if fake mode is activated. That creates the bug because constant tensors do need their value in initializers.

Reply:
If you don't like the setter, I guess the other way around would be we change `include_initializers` to something like `filtered_initializers`, and passed in with `to_model_proto` API.

Reply:
Or I guess we could add another parameter: fake_tensor: bool in `add_initializer`.